pkg/containerutil/containerutil.go

```diff
@@ -129,3 +130,48 @@ func WithBindMountHostProcfs(_ context.Context, _ oci.Client, _ *containers.Cont
 	s.Linux.ReadonlyPaths = newROP
 	return nil
 }
+
+// GenerateSharingPIDOpts returns the oci.SpecOpts for shared container
+// that is in host namespace running inside rootless mode or not
+// or in linux namespace.
```

Comment:
Sorry I don't quite understand this part 

```
// that is in host namespace running inside rootless mode or not
// or in linux namespace.`.
```

If I read the func correctly, is it doing something like this?
```
// GenerateSharingPIDOpts returns the oci.SpecOpts that shares the host linux namespace from `targetCon` 
// If `targetCon` doesn't have a `PIDNamespace`, a new one is generated from its `Pid`)
```